mrpeek: minor optimisation for sixel encoding

jdtournier opened the pull request:
Based on recent discussion with Guillaume Flandin on OHBM Mattermost channel, and [this site](https://qiita.com/arakiken/items/4a216af6547d2574d283) he alerted me to. 

Seems our encoding is about as efficient as it could be. These modifications reduce the frame size for a 256×244 grayscale image from 72kb to 65kb. Not enormously exciting, but as it was simple to implement...

jdtournier committed:
mrpeek: minor optimisation for sixel encoding

dchristiaens committed:
Merge branch 'master' into minor_optimisations_for_sixel_encoding

dchristiaens commented:
Hmm, this one seems to need more testing. Not sure what's going on here:
<img width="895" alt="Screenshot 2020-06-19 at 19 22 18" src="https://user-images.githubusercontent.com/7466983/85168507-5d4ba880-b262-11ea-9c76-c4454c57dd6e.png">

jdtournier commented:
Interesting. I've no idea how that can even happen... 

Does this happen when the image is larger than 256 pixels? In which case I might need to tweak that `commit()` method to split repeating sequences up into chunks of at most 256 repeats (or whatever the maximum supported is).

jdtournier commented:
And I assume this is on iTerm2?

dchristiaens commented:
Yes, iTerm2.

The same thing happens when the frame is smaller than 256 pixels.

jdtournier commented:
any chance you could run that with the `-noninteractive` option and redirect to file, and send it over so I can take a look...?

jdtournier committed:
remove write background 'optimisation'

It doesn't seem to play well with the 'skip unchanged at end of line'
optimisation...

jdtournier commented:
OK, I managed to reproduce something that kinds looked a bit like that. I removed one of the optimisations, hopefully that'll fix it...

maxpietsch commented:
Looks good to me. There seems to be a small black bar below sixels (mlterm and iTerm2):
![image](https://user-images.githubusercontent.com/10046944/85181606-a3b2fe80-b286-11ea-8b26-9ce5161d72f6.png) but it's already in `master`

dchristiaens reviewed (approved):
That seems to have done the trick.

jdtournier commented:
There's a chance that black line might be fixed by changing the [SixelStart sequence](https://github.com/MRtrix3/mrpeek/blob/master/src/vt_control.h#L14) to `"\033P;1;;q$"`...

jdtournier commented:
OK, forget that last idea. The issue is that the buffer holding the data prior to encoding is zero-filled and its height is set to a multiple of 6. So it does get set to actual zero, rather than remaining unset as should be the case. Not related to this PR, so might be best to deal with that as a separate issue?
